<a href="https://colab.research.google.com/github/olaqarawani/Data-Science/blob/main/Assignment1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Ola Baninemra 1210967

In [ ]:
import pandas as pd

In [ ]:
data= pd.read_csv('C:\\Users\\sef\\Downloads\\assignment_1-python.csv')

In [ ]:
df = pd.DataFrame(data)

**Q1) Identify the most expensive vegetables and fruits commodity during this period.**


In [ ]:
fruits = df[df['category'] == 'vegetables and fruits']
most_expensive = fruits[fruits['price'] == fruits['price'].max()]

result = most_expensive[['commodity', 'price']]
result

# max_values =df.apply(max,axis=0)
# max_values

,commodity,price
168,Apples (red),6.09


The most expensive vegetable and fruits commodities are displayed above based on the price.



**Q2) Add a new column named currency_rate that captures the exchange rate between the local currency and the US dollar (USD).**


In [ ]:
df['currency_rate'] = df['price'] / df['usdprice']
df.head(10)


,date,region,city,market,latitude,longitude,category,commodity,unit,priceflag,pricetype,currency,price,usdprice,currency_rate
0,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,cereals and tubers,Bread,KG,actual,Retail,ILS,2.50,0.7934,3.150996
1,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,cereals and tubers,Potatoes (medium size),KG,actual,Retail,ILS,1.94,0.6157,3.150885
2,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,cereals and tubers,"Rice (small grain, imported)",25 KG,actual,Retail,ILS,152.31,48.3351,3.151126
3,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,cereals and tubers,Wheat flour (locally processed),50 KG,actual,Retail,ILS,90.00,28.5612,3.151128
4,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,"meat, fish and eggs",Eggs,2 KG,actual,Retail,ILS,11.72,3.7193,3.151131
5,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,"meat, fish and eggs",Fish (frozen),KG,actual,Retail,ILS,11.33,3.5955,3.151161
6,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,"meat, fish and eggs",Meat (beef),KG,actual,Retail,ILS,38.50,12.2178,3.151140
7,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,"meat, fish and eggs",Meat (chicken),KG,actual,Retail,ILS,14.85,4.7126,3.151127
8,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,"meat, fish and eggs","Meat (goat, with bones)",KG,actual,Retail,ILS,54.56,17.3144,3.151134
9,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,milk and dairy,Cheese (goat),KG,actual,Retail,ILS,16.57,5.2584,3.151149


I calculated the currency_rate by dividing the price by the usdprice

**Q3) Create a new column called unified_price that represents the price in USD per kilogram for commodities measured by weight and the price per liter for commodities measured by volume.**

In [ ]:
def unified_price(n):
    price = n['usdprice']

    if 'KG' in n['unit']:
        return f"{price}/kg"
    elif 'L' in n['unit']:
        return f"{price}/L"
    elif 'G' in n['unit']:
        return f"{price * 1000}/kg"
    elif 'Cubic meter' in n['unit']:
        return f"{price / 1000}/L"

    return price

df['unified_price'] = df.apply(unified_price, axis=1)
df.head(20)

,date,region,city,market,latitude,longitude,category,commodity,unit,priceflag,pricetype,currency,price,usdprice,currency_rate,unified_price
0,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,cereals and tubers,Bread,KG,actual,Retail,ILS,2.50,0.7934,3.150996,0.7934/kg
1,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,cereals and tubers,Potatoes (medium size),KG,actual,Retail,ILS,1.94,0.6157,3.150885,0.6157/kg
2,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,cereals and tubers,"Rice (small grain, imported)",25 KG,actual,Retail,ILS,152.31,48.3351,3.151126,48.3351/kg
3,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,cereals and tubers,Wheat flour (locally processed),50 KG,actual,Retail,ILS,90.00,28.5612,3.151128,28.5612/kg
4,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,"meat, fish and eggs",Eggs,2 KG,actual,Retail,ILS,11.72,3.7193,3.151131,3.7193/kg
5,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,"meat, fish and eggs",Fish (frozen),KG,actual,Retail,ILS,11.33,3.5955,3.151161,3.5955/kg
6,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,"meat, fish and eggs",Meat (beef),KG,actual,Retail,ILS,38.50,12.2178,3.151140,12.2178/kg
7,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,"meat, fish and eggs",Meat (chicken),KG,actual,Retail,ILS,14.85,4.7126,3.151127,4.7126/kg
8,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,"meat, fish and eggs","Meat (goat, with bones)",KG,actual,Retail,ILS,54.56,17.3144,3.151134,17.3144/kg
9,2022-01-15,Gaza Strip,Kan Younis,Gaza Strip,31.34611,34.32573,milk and dairy,Cheese (goat),KG,actual,Retail,ILS,16.57,5.2584,3.151149,5.2584/kg


In this question when the unit was 'G' I converted it to kg by multiplying by 1000, and when it was 'Cubic meter' I converted it to L by dividing by 1000

**Q4) Using the unified_price, find the maximum price reached during this period and for which commodity for both:**

- commodities measured by weight

- commodities measured by volume

In [ ]:
weight = df[df['unit'].str.contains('KG|G')]
max_weight = weight.loc[weight['unified_price'] == weight['unified_price'].max()]
result_weight = max_weight[['commodity', 'unified_price']]

volume = df[df['unit'].str.contains('L|Cubic meter')]
max_volume= volume.loc[volume['unified_price'] == volume['unified_price'].max()]
result_volume = max_volume[['commodity', 'unified_price']]

result_weight, result_volume




(      commodity unified_price
 20  Oil (olive)     9.5204/kg,
        commodity unified_price
 342  Oil (maize)       9.576/L)

I used contains to check if the unit includes KG as some commodities have numerical values, like Milk (powder) with a unit of 2.5 KG. The same applies to L or Cubic meter.